### PR TITLE
[refine](pipeline) Move the statistics of the sink and source into the basic class.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run cloud_p0

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/local_exchange/local_exchanger.h [hunk 1]
```diff
@@ -61,8 +66,10 @@ class ExchangerBase {
     friend class LocalExchangeSourceLocalState;
     friend class LocalExchangeSinkOperatorX;
     friend class LocalExchangeSinkLocalState;
-    std::atomic<int> _running_sink_operators = 0;
-    std::atomic<int> _running_source_operators = 0;
+    // ref from LocalExchangeSharedState
+    const std::atomic<int>& _running_sink_operators;
```
Comment (Gabriel39) [inline be/src/pipeline/local_exchange/local_exchanger.h:70]: Just delete this

File: be/src/pipeline/dependency.h [hunk 2]
```diff
@@ -72,16 +70,33 @@ struct BasicSharedState {
...
         return reinterpret_cast<const TARGET*>(this);
     }
-    std::vector<DependencySPtr> source_deps;
-    std::vector<DependencySPtr> sink_deps;
+    //sub when operator close
+
+    virtual void sub_running_sink_operators();
+    // When all sources are closed, it will wake up the sink
+    // (for example, if there is a limit in the downstream pipeline, causing the source to end early).
+    virtual void sub_running_source_operators();
```
Comment (Gabriel39) [inline be/src/pipeline/dependency.h:78]: remove `virtual`

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run cloud_p1

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Pull request closed without merging**